src/zebra_puzzles/zebra_utils.py

```diff
@@ -59,33 +27,81 @@ def generate_solution(
         sample(list(attributes[cat].keys()), k=n_objects) for cat in chosen_categories
     ]
 
-    # Add the object indices to the solution
-    indices = [str(x) for x in list(range(n_objects))]
-    solution = [indices] + chosen_attributes
+    # Transpose the attribute matrix
+    chosen_attributes = [
+        [row[i] for row in chosen_attributes] for i in range(n_attributes)
+    ]
+
+    solution = [[str(i + 1)] + row for i, row in enumerate(chosen_attributes)]
 
     return solution, chosen_categories, chosen_attributes
 
 
-def choose_clues(
-    solution: List[List],
-    clues: List,
-    chosen_categories: List,
+def save_dataset(data: str, filename: str, folder: str = "data") -> None:
+    """Save a zebra puzzle dataset.
+
+    Args:
+        data: Data to save.
+        filename: Name of the file.
+        folder: Folder to save the file in.
+
+    TODO: Consider preferred format.
+    """
+    with open(folder + "/" + filename, "w") as file:
+        file.write(data)
+
+
+def complete_prompt(
+    chosen_clues: List[str],
+    n_objects: int,
+    chosen_categories: List[str],
     chosen_attributes: List[List],
+    prompt_template: str,
 ) -> str:
-    """Generate a zebra puzzle.
+    """Complete the prompt with the chosen clues.
+
+    Assumes commas are used similarly across the supported languages.
 
     Args:
-        solution: Solution to the zebra puzzle as a list of lists representing the solution matrix of object indices and chosen attribute values. This matrix is n_attributes x n_objects.
-        clues: Possible clues to include in the clues as a list of tuples. Each tuple contains the clue name and function. TODO: Edit this description when the clues are implemented.
+        chosen_clues: Chosen clues for the zebra puzzle as a list of strings.
+        n_objects: Number of objects in the puzzle.
         chosen_categories: Categories chosen for the solution.
         chosen_attributes: Attribute values chosen for the solution.
+        prompt_template: Template for the prompt.
 
     Returns:
-        chosen_clues: Clues for the zebra puzzle as a string.
+        prompt: The full prompt for the zebra puzzle as a string.
+
 
-    TODO: Implement the generation of the clues.
+    TODO: Improve the prompt here and in the config file.
+    TODO: Add support for puzzles with a single category and/or object
     """
-    chosen_clues = (
-        "1. This is an example. \n 2. This is the second part of the example."
+    chosen_clues = [f"{i + 1}. {clue}" for i, clue in enumerate(chosen_clues)]
+
+    if len(chosen_clues) > 1:
+        chosen_clues_str = "\n".join(chosen_clues)
+    else:
+        chosen_clues_str = chosen_clues[0]
+
+    chosen_categories_part1 = ", ".join(chosen_categories[:-1])
+    chosen_categories_part2 = chosen_categories[-1]
+
+    # Transpose chosen_attributes
+    chosen_attributes = list(map(list, zip(*chosen_attributes)))
```

Comment:
This kind of thing would be a lot easier if `chosen_attributes` was a NumPy array: simply `chosen_attributes = chosen_attributes.T`.